medusa/providers/torrent/json/torrentday.py

```diff
@@ -0,0 +1,174 @@
+# coding=utf-8
+
+"""Provider code for TorrentDay."""
+
+from __future__ import unicode_literals
+
+import logging
+import re
+from medusa import tv
+from medusa.helper.common import convert_size
+from medusa.logger.adapters.style import BraceAdapter
+from medusa.providers.torrent.torrent_provider import TorrentProvider
+from requests.compat import urljoin
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class TorrentDayProvider(TorrentProvider):
+    """TorrentDay Torrent provider."""
+
+    def __init__(self):
+        """Initialize the class."""
+        super(TorrentDayProvider, self).__init__('TorrentDay')
+
+        # URLs
+        self.url = 'https://www.torrentday.com'
+        self.urls = {
+            'login': urljoin(self.url, '/torrents/'),
+            'search': urljoin(self.url, '/t.json'),
+            'download': urljoin(self.url, '/download.php/')
+        }
+
+        # Proper Strings
+
+        # Miscellaneous Options
+        self.freeleech = False
+        self.enable_cookies = True
+        self.cookies = ''
+        self.required_cookies = ('uid', 'pass')
+
+        # TV/480p - 24
+        # TV/Bluray - 32
+        # TV/DVD-R - 31
+        # TV/DVD-Rip - 33
+        # TV/Mobile - 46
+        # TV/Packs - 14
+        # TV/SD/x264 - 26
+        # TV/x264 - 7
+        # TV/x265 - 34
+        # TV/XviD - 2
+
+        self.categories = {
+            'Season': {'14': 1},
+            'Episode': {'2': 1, '26': 1, '7': 1, '24': 1, '34': 1},
+            'RSS': {'2': 1, '26': 1, '7': 1, '24': 1, '34': 1, '14': 1}
+        }
+
+        # Torrent Stats
+        self.minseed = None
+        self.minleech = None
+
+        # Cache
+        self.cache = tv.Cache(self, min_time=10)  # Only poll IPTorrents every 10 minutes max
+
+    def search(self, search_strings, age=0, ep_obj=None, **kwargs):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+        if not self.login():
+            return results
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+                if mode != 'RSS':
+                    log.debug('Search string: {search}',
+                              {'search': search_string})
+
+                search_string = '+'.join(search_string.split())
+
+                params = dict({'q': search_string}, **self.categories[mode])
+
+                if self.freeleech:
+                    params.update({'free': 'on'})
```

Comment:
I'm not sure if this still has any effect?

Reply:
afaik its to only list free leech torrents. @duramato confirm?

Reply:
Yeah but is it available through the new api?